### ISSUE TITLE
Multi-valued rel Attribute

### ISSUE THREAD

**Issue by ibnesayeed** [thlorenz/parse-link-header#6]
Like "class" attribute in HTML elements, "rel" attribute of the link header can have multiple space separated values e.g., `rel="next last"` which I think will not be properly treated by the current code:

```js
var m = p.match(/\s*(.+)\s*=\s*"?([^"]+)"?/)
if (m) acc[m[1]] = m[2];
```

**Comment by jarofghosts** [comment]
i just wrote a test for this and it passed. can you verify?

**Comment by ibnesayeed** [comment]
Here is a quick test case:

```js
var parse = require('parse-link-header');
var linkHeader = '<https://api.github.com/user/9287/repos?page=3&per_page=100>; rel="next last"';
var parsed = parse(linkHeader);
console.log(parsed);

// Gives following result =>
{ 'next last': 
   { page: '3',
     per_page: '100',
     rel: 'next last',
     url: 'https://api.github.com/user/9287/repos?page=3&per_page=100' } }

// While the expected output is =>
{ next: 
   { page: '3',
     per_page: '100',
     rel: 'next',
     url: 'https://api.github.com/user/9287/repos?page=3&per_page=100' },
 last: 
   { page: '3',
     per_page: '100',
     rel: 'last',
     url: 'https://api.github.com/user/9287/repos?page=3&per_page=100' }}
```

**Comment by jarofghosts** [comment]
that's actually not what i would expect

**Comment by ibnesayeed** [comment]
I am not sure what you would expect, but what I illustrated is in accordance with the [RFC 5988: Web Linking](https://tools.ietf.org/html/rfc5988#section-5.5). It allows presenting multiple relations together if the linked resource is related to the current resource in more than one ways. This is an optional syntax, but perfectly valid and space separated relations in the value field should be treated as separate relations. There is one exception to this rule that I know which is due to the extension in HTML4 <link> element relation where if the two relations "alternate stylesheet" coincide, they are treated together.

**Comment by jarofghosts** [comment]
Ah, sorry I think I misunderstood the original case. I can look into supporting this

**Comment by thlorenz** [comment]
Fixed in 0.4.
